#!/bin/bash

# Define variables
samplesTSV="/add/results/2024-03-04_force_mutect/samples.tsv"
fastaFile="/home/mcarta/databases/hg38.fa"
mutationName="E200K"
outputDir="/add/results/2024-03-04_force_mutect/2024-03-10_E200K_VCFtest/"

###test

# output file (generated by Mutect2)
#mutationVCF="/add/variant_VCF/E200K.vcf.gz"

# manual file with sample and FORMAT data (I made it)
# mutationVCF="/add/variant_VCF/E200K_manual_test.vcf.gz"

# manual file, also with headers (based on other manual one)
mutationVCF="/add/variant_VCF/E200K_manual_test.vcf.gz"



###

inputFile="/add/seq_data/2021-10-19_first_CJD_seq/20211027.0-o26424_1_1-CJD1_fr.picard.markedDup.recal.bam"


    # Perform Mutect2 analysis for each sample
    gatk Mutect2 \
        -R "$fastaFile" \
        --alleles "$mutationVCF" \
        --genotype-germline-sites true \
        -L "$mutationVCF" \
        -I "$inputFile" \
        -O "${outputDir}/${mutationName}_test_germlineoption.vcf"

