\documentclass[a4paper]{article}
\usepackage[margin=1.5cm]{geometry}
\usepackage{adjustbox}
\usepackage{afterpage}  % \afterpage
\usepackage{amsmath}    % \text{} in math mode
\usepackage{array}
\usepackage{booktabs}
\usepackage{caption}
\usepackage{colortbl}
\usepackage{graphicx}
\usepackage{makecell}
\usepackage{multirow}
\usepackage{pdflscape}  % landscape pages that rotate correctly in PDF viewers
\usepackage{threeparttable}
\usepackage[table]{xcolor} % for \rowcolor and named colours like gray
\usepackage{siunitx}
\usepackage{xspace}     % insert space after macros

% Column type used in tabular:
% 'L' = left-aligned math mode column
% 'C' = centered matho mode
\newcolumntype{L}{>{$}l<{$}}
\newcolumntype{C}{>{$}c<{$}}

% macros
\newcommand{\prnp}{\textit{PRNP}\xspace}
\newcommand{\prpsc}{PrP\textsuperscript{Sc}\xspace}
\newcommand{\ra}[1]{\renewcommand{\arraystretch}{#1}}
\newcommand{\xfold}{\ensuremath{\times}}

% ---- PRNP variant shorthands ----
\newcommand{\dn}{D178N\xspace}
\newcommand{\ek}{E200K\xspace}
\newcommand{\pl}{P102L\xspace}

% Struts used in patients table for vertical spacing
\newcommand{\Tstrut}{\rule{0pt}{2.6ex}}
\newcommand{\Bstrut}{\rule[-1.2ex]{0pt}{4.0ex}}


\captionsetup[table]{skip=6pt,font=footnotesize,justification=raggedright,singlelinecheck=false}
\renewcommand{\arraystretch}{1.18}

\sisetup{
	round-mode          = places,
	round-precision     = 2,
	table-align-text-post = false,
	group-minimum-digits = 4,
	group-separator     = {\,}
}

\captionsetup[table]{skip=5pt, font=footnotesize, justification=raggedright, singlelinecheck=false}
\renewcommand{\arraystretch}{1.2} % Slightly more vertical space



\begin{document}


% patient table


%patient list
\afterpage{%
	\clearpage% Flush earlier floats (otherwise order might not be correct)
	\thispagestyle{empty}% empty page style
	\begin{landscape}% Landscape page
		\centering % Center table
		\resizebox{1\linewidth}{!}{%
			\begin{tabular}{lCcClllcll} %age and duration in math mode. Put NA in \text{}.
				\toprule[2pt]\Bstrut
				name & \text{age} & sex & \text{time to death (d)} & symptoms & main diagnosis & other diagnoses & codon $129$ & histotype & \prpsc type \\
				\hline \Tstrut
				CJD1 & 66 & F & 95 & visual disturbances, ataxia, dystonia, EEG compatible with sCJD & sCJD &  & MM & MM1+2C & type 1 (2) \\
				CJD2 & 74 & M & 74 & mild dementia, myoclonus, nystagmus, ataxia, palsy of right arm & sCJD &  & MM & MM1 & type 1 \\
				CJD3 & 69 & M & 368 & dementia (cortical and subcortical) & sCJD &  & MM & atypical & type 2 \\
				CJD4 & 66 & F &  &  & sCJD &  & VV & VV2 & type 2 \\
				CJD5 & 71 & M & 543 & progressive fronto-subcortical dysfunction, visual disturbances, EEG compatible with sCJD & sCJD &  & MV & MV2K+C & type 2 (1) \\
				CJD6 & 62 & F &  &  & sCJD &  & MV & MV2C & type 2 \\
				CJD7 & 78 & F & 324 & moderate cognitive deficits, postural instability, sensory axonal polyneuropathy & sCJD &  & MV & atypical & type 1 \\
				CJD8 & 72 & M & 169 & severe cognitive deficits, memory loss, visual disturbances, myoclonus, ataxia, gait disorder & sCJD &  & MM & MM1 & type 1 \\
				CJD9 & 56 & M & 86 & cognitive deficits, memory loss, amnestic aphasia, non-convulsive status epilepticus & sCJD &  & MM & atypical & type 1 \\
				CJD10 & 73 & F & 121 & Broca's aphasia, mild cerebellar dysfunction & sCJD & intermed. AD, mild CAA & MM & MM1+2C & type 1 (2) \\
				CJD11 & 48 & F & 64 & memory loss, affective disorder, nystagmus, extrapyramidal symptoms, myoclonus & sCJD &  & MM & MM1 & type 1 \\
				CJD12 & 76 & M & 213 & rapidly progressive dementia, visual disturbances, gait disorder & sCJD &  & MM & MM1 & type 1 \\
				CJD13 & 57 & F & 100 & cognitive deficits, memory loss, amnestic aphasia, gait disorder & sCJD &  & MM & MM1 & type 1 \\
				CJD14 & 83 & M & 152 & dementia, myoclonus, visual disturbances, positive psychiatric symptoms & sCJD &  & MV & atypical & type 1 \\
				CJD15 & 63 & M & 95 & dementia, visual disturbances, cerebellar dysfunction, pyramidal signs, extrapyramidal symptoms & sCJD & low AD, moderate CAA & MM & MM+2C & type 1 \\
				CJD16 & 71 & F & 55 & mild cognitive deficits (loss of memory, concentration, impulse control), myoclonus, ataxia & sCJD & low AD & MM & MM1 & type 1 \\
				CJD17 & 67 & M &  &  & sCJD & low AD & MM & MM1+2C & type 1 \\
				CJD18 & 69 & F & 180 & myoclonus, pyramidal signs & sCJD & low AD & MV & MV2K+C & type 2 \\
				CJD19 & 59 & M & 363 & dementia, progressive dysarthria & sCJD & stroke & MV & atypical & types 1 \& 2 \\
				CJD20 & 57 & M & 156 & dementia, myoclonus, cerebellar dysfunction & sCJD &  & MM & MM1 & type 1 \\
				CJD21 & 69 & M & 131 & rapidly progressive dementia, ataxia, myoclonus, EEG compatible with sCJD & sCJD & low AD, moderate CAA & MM & MM1 & type 1 \\
				CJD22 & 72 & M & 55 & dysarthria, dysphagia, gait disorder & sCJD & PART, schizophrenia & MM & MM1 & type 1 \\
				CJD23 &  &  &  &  & sCJD &  & VV & VV1 & \\
				CJD24 &  &  &  &  & sCJD &  & MM & MM2T & \\
				CJD25 &  &  &  &  & sCJD &  & MM & MM2T & \\
				CJD26 &  &  &  &  & sCJD &  & MM & MM2T & \\
				CJD27 &  &  &  &  & sCJD &  & MV & MV2K & \\
				CJD28 &  &  &  &  & sCJD &  & MV & MV2K & \\
				CJD29 &  &  &  &  & sCJD &  & MV & MV2K & \\
				CJD30 & 67 &  &  &  & gCJD (E200K heterozygous) &  & VV & VV2 & \\
				CJD31 & 74 &  &  &  & sCJD &  & VV & VV2 & \\
				\hline \Tstrut
				Control1 & 78 & F &  & dementia, Parkinsonism, multifocal epilepsy, EEG compatible with sCJD & high AD & CAA & MV &  &  \\
				Control2 & 72 & M &  & rapidly progressive neurological disorder, rigidity & multiple ischaemic strokes & atherosclerosis, PART & MV &  &  \\
				Control3 & 76 & M &  & rapidly progressive dementia, postural instability, cachexia & CMV encephalitis &  & MM &  &  \\
				Control4 & 78 & M &  & visual disturbances, dizziness, paralysis of eye muscles & severe alpha-synucleinopathy & PART & MM &  &  \\
				Control5 & 70 & F &  & dementia, Parkinsonism, temporal lobe atrophy & Lewy body disease & PART, TDP-43 & MV &  &  \\
				Control6 & 84 & F &  & ataxia, reduced vigilance, epilepsy, EEG compatible with sCJD & paraneoplastic encephalomyelitis & low AD & MM &  &  \\
				Control7 & 64 & M &  & dementia & FTLD with TDP-43 pathology & intermed. AD & MV &  &  \\
				Control8 & 78 & F &  & rapidly progressive dementia, cerebellar dysfunction, pyramidal signs, epilepsy & CAA & inflammatory encephalopathy & MV &  &  \\
				\bottomrule[2pt]
			\end{tabular}
		}
		\captionof{table}{\textbf{Overview of patients included in the study}. Age at death, sex, time from onset of symptoms to death (illness duration, given in days), main clinical symptoms, main neuropathological diagnosis, other neuropathological diagnoses, \prnp codon $129$ genotype, prion disease histotype and predominant \prpsc subtype (minor \prpsc subtype given in brackets, if present) are given, where available. In the control samples, other brain pathologies were identified. Sample CJD30 was unexpectedly found to be germline heterozygous for the E200K variant, classifying it as genetic CJD (gCJD). Low, intermediate, high AD: degree of AD-related neuropathological change. AD: Alzheimer's disease. CAA: cerebral amyloid angiopathy. CMV: cytomegalovirus. FTLD: frontotemporal lobar degeneration. PART: primary age-related tauopathy.}
		\label{tab:patients}
	\end{landscape}
	\clearpage% Flush page
}

\clearpage

% ddPCR LoD table
	
\begin{table}[t]
	\centering
	\begin{adjustbox}{max width=\textwidth}
		\begin{minipage}{\textwidth}
			\centering
			{\normalsize
		
			\begin{tabular}{@{} l r r r r r @{}}
				\toprule
				Sample & Expected & Measured (95\% CI) & Droplets & Positive & Negative \\
				\midrule
				\addlinespace
				\textbf{D178N} & & & & & \\
				NTC       & 0.00 & 0.00 (0.00--0.00) & 7092  & 0   & 7092 \\
				WT        & 0.00 & 0.01 (0.00--0.02) & 13062 & 3   & 13059 \\
				0.05\%    & 0.05 & 0.06 (0.03--0.08) & 13726 & 18  & 13708 \\
				0.1\%     & 0.10 & 0.17 (0.15--0.20) & 46272 & 178 & 46094 \\
				0.33\%    & 0.33 & 0.31 (0.27--0.36) & 23707 & 156 & 23551 \\
				1\%       & 1.00 & 1.21 (1.11--1.30) & 22879 & 601 & 22278 \\
				\addlinespace
				\textbf{E200K} & & & & & \\
				NTC       & 0.00 & 0.00 (0.00--0.00) & 23746 & 0   & 23746 \\
				WT        & 0.00 & 0.01 (0.00--0.02) & 25239 & 6   & 25233 \\
				0.05\%    & 0.05 & 0.07 (0.05--0.08) & 53456 & 86  & 53370 \\
				0.1\%     & 0.10 & 0.12 (0.10--0.14) & 53203 & 149 & 53054 \\
				0.33\%    & 0.33 & 0.33 (0.28--0.38) & 25176 & 174 & 25002 \\
				1\%       & 1.00 & 1.06 (0.97--1.15) & 24811 & 532 & 24279 \\
				\addlinespace
				\textbf{P102L} & & & & & \\
				NTC       & 0.00 & 0.00 (0.00--0.00) & 29652 & 0   & 29652 \\
				WT        & 0.00 & 0.07 (0.04--0.09) & 22130 & 23  & 22107 \\
				0.05\%    & 0.05 & 0.13 (0.11--0.15) & 59896 & 139 & 59757 \\
				0.1\%     & 0.10 & 0.18 (0.16--0.21) & 58993 & 174 & 58819 \\
				0.33\%    & 0.33 & 0.49 (0.43--0.55) & 28280 & 257 & 28023 \\
				1\%       & 1.00 & 1.42 (1.31--1.53) & 27914 & 643 & 27271 \\
				\bottomrule
			\end{tabular}
			\vspace{0.5em}
			
				\caption{\textbf{ddPCR quantification of mutant allele fractions in dilution series and controls.}
				Block headers indicate the assayed \textit{PRNP} mutation. Controls (NTC, WT) are included.
				“Expected” is the nominal mutant fraction in the input mixture. “Measured (95\% CI)” gives the
				mutant allele fraction with two-sided 95\% Poisson CIs (QuantaSoft Analysis Pro). Droplet counts are after quality filtering.}
			\label{tab:ddpcr_fractions}
		}
		\end{minipage}
	\end{adjustbox}
\end{table}

\clearpage

\begin{table}[t]
	\centering
	\resizebox{\textwidth}{!}{
		\begin{tabular}{%
				l  % Sample
				l  % Variant
				l  % dbSNP ID
				l  % Chromosome
				r  % Position
				c  % REF
				c  % ALT
				l  % Mutation Type
				r  % Read Depth (DP)
				r  % REF Count
				r  % ALT Count
				r  % AAF (\%)
				r  % Base Quality
				r  % Mapping Quality
				r  % REF Forward Count
				r  % REF Reverse Count
				r  % ALT Forward Count
				r  % ALT Reverse Count
			}
			\toprule
			Sample & Variant & dbSNP ID & Chromosome & Position & REF & ALT 
			& Mutation Type & Read Depth (DP) & REF Count & ALT Count 
			& AAF (\%) & Base Quality & Mapping Quality 
			& REF Forward Count & REF Reverse Count 
			& ALT Forward Count & ALT Reverse Count \\
			\midrule
			A117V 0.5\% spike-in & A117V & rs74315402 & chr20 & 4699570 
			& C & T & MISSENSE & 6961 & 6869 & 92 
			& 1.321649 & 60 & 21.88 
			& 3730 & 3139 & 40 & 52 \\
			A117V 1\% spike-in & A117V & rs74315402 & chr20 & 4699570 
			& C & T & MISSENSE & 4315 & 4280 & 35 
			& 0.811124 & 60 & 22.24 
			& 2216 & 2064 & 16 & 19 \\
			\bottomrule
		\end{tabular}
	}
	\vspace{0.5em}
	\caption{\textbf{A117V spike-in analysis for limit-of-detection determination.} Sequencing metrics for A117V (rs74315402) spike-in mixtures at nominal 0.5\% and 1\% alternate-allele frequencies. Both variants were confidently detected by Mutect2 and passed all quality filters. Observed variant-allele fractions (AAFs) were 1.32\% and 0.81\%, respectively, with balanced strand support and high base quality.}
\end{table}

\clearpage


% SNV table

\begin{table}[t]
	\centering
	{\normalsize
		\begin{adjustbox}{max width=\textwidth}
			\begin{tabular}{>{\centering\arraybackslash}p{1.3cm}  % Sample
					>{\centering\arraybackslash}p{1.9cm}  % Chromosome
					>{\centering\arraybackslash}p{1.7cm}  % Position
					>{\centering\arraybackslash}p{1.1cm}  % AAF
					>{\centering\arraybackslash}p{0.7cm}  % REF
					>{\centering\arraybackslash}p{0.7cm}  % ALT
					>{\centering\arraybackslash}p{1.6cm}  % Mutation Type
					>{\centering\arraybackslash}p{2.2cm}  % Pop Freq
					>{\centering\arraybackslash}p{1.2cm}  % Read Depth
					>{\centering\arraybackslash}p{1.3cm}  % REF Count
					>{\centering\arraybackslash}p{1.3cm}  % ALT Count
					>{\centering\arraybackslash}p{1.4cm}  % Base Qual
					>{\centering\arraybackslash}p{1.4cm}  % Mapping Q
					>{\centering\arraybackslash}p{1.0cm}  % REF Fwd
					>{\centering\arraybackslash}p{1.0cm}  % REF Rev
					>{\centering\arraybackslash}p{1.0cm}  % ALT Fwd
					>{\centering\arraybackslash}p{1.0cm}} % ALT Rev
				\toprule
				\textbf{Sample} & \textbf{Chromosome} & \textbf{Position} & \textbf{AAF (\%)} & \textbf{REF} & \textbf{ALT} & \textbf{Mutation Type} & \textbf{Population Frequency} & \textbf{Read Depth} & \textbf{REF Count} & \textbf{ALT Count} & \textbf{Base Quality} & \textbf{Mapping Quality} & \textbf{REF Fwd} & \textbf{REF Rev} & \textbf{ALT Fwd} & \textbf{ALT Rev} \\
				\midrule
				\rule{0pt}{1.2em}CJD2  & Chr20 & 4691920 & 0.55 & G & A & intron & not described & 4706 & 4680 & 26 & 37.9 & 22.01 & 2327 & 2353 & 10 & 16 \\
				\rule{0pt}{1.2em}CJD6  & Chr20 & 4691920 & 0.52 & G & A & intron & not described & 3482 & 3464 & 18 & 33.4 & 20.10 & 1787 & 1677 & 6  & 12 \\
				\rule{0pt}{1.2em}CJD23 & Chr20 & 4691920 & 0.60 & G & A & intron & not described & 5133 & 5102 & 31 & 38.5 & 21.00 & 2525 & 2577 & 12 & 19 \\
				\rule{0pt}{1.2em}CJD23 & Chr20 & 4694249 & 0.82 & T & C & intron & not described & 1344 & 1333 & 11 & 30.0 & 20.62 & 686  & 647  & 5  & 6 \\
				\bottomrule
			\end{tabular}
		\end{adjustbox}
	}
	
	\vspace{0.5em}
	
	% SNV table
	
	\begin{adjustbox}{max width=\textwidth}
		\begin{minipage}{\linewidth}
			\caption{\textbf{Possible somatic single nucleotide variants (sSNVs)} detected in the \textit{PRNP} gene. A total of four intronic variants were identified: one variant each in samples CJD2 and CJD6, and two variants in sample CJD23. The alternative allele fraction (AAF) ranged from 0.52\% to 0.82\%. Abbreviations: AAF, alternative allele fraction; REF, reference allele; ALT, alternative allele; Fwd, forward read count; Rev, reverse read count.}
		\end{minipage}
	\end{adjustbox}
	
\end{table}

\clearpage

	% ddPCR results table
\begin{table}[p]
	\ra{1.0}
	\centering
	\begin{adjustbox}{max width=\linewidth, max totalheight=0.95\textheight}
		\begin{tabular}{lLLLLLLLL}
			\toprule[2pt]
			\multirow{2}{*}{\text{name}} &
			\multicolumn{4}{c}{\text{Analysed (n)}} &
			\multicolumn{4}{c}{\text{Max. VAF}}\\
			\cmidrule(lr){2-5}\cmidrule(lr){6-9}
			& \text{samples} & \text{\dn} & \text{\ek} & \text{\pl} & \text{\dn} & \text{\ek} & \text{\pl} & \text{overall}\\
			\midrule \Tstrut
			
			CJD1 & 5 & 5 & 5 & 5 & 0.015 & 0.046 & 0.032 & 0.046\\
			CJD2 & 6 & 6 & 6 & 6 & 0.067 & 0.033 & 0.044 & 0.067\\
			CJD3 & 6 & 6 & 6 & 6 & 0.047 & 0.045 & 0.054 & 0.054\\
			CJD4 & 5 & 5 & 5 & 5 & 0.046 & 0.009 & 0.049 & 0.049\\
			CJD5 & 6 & 6 & 6 & 6 & 0.042 & 0.022 & 0.043 & 0.043\\
			CJD6 & 6 & 6 & 6 & 6 & 0.023 & 0.065 & 0.056 & 0.065\\
			CJD7 & 6 & 6 & 6 & 6 & 0.020 & 0.021 & 0.058 & 0.058\\
			CJD8 & 6 & 6 & 6 & 6 & 0.052 & 0.044 & 0.168 & 0.168\\
			CJD9 & 6 & 6 & 6 & 6 & 0.033 & 0.030 & 0.047 & 0.047\\
			CJD10 & 6 & 6 & 6 & 6 & 0.043 & 0.032 & 0.086 & 0.086\\	
			CJD11 & 6 & 6 & 6 & 6 & 0.035 & 0.022 & 0.045 & 0.045\\
			CJD12 & 6 & 6 & 6 & 6 & 0.040 & 0.051 & 0.046 & 0.051\\
			CJD13 & 6 & 6 & 6 & 6 & 0.073 & 0.078 & 0.034 & 0.078\\
			CJD14 & 6 & 6 & 6 & 6 & 0.033 & 0.035 & 0.001 & 0.035\\
			CJD15 & 6 & 6 & 6 & 6 & 0.035 & 0.042 & 0.131 & 0.131\\	
			CJD16 & 1 & 1 & 1 & 1 & 0.054 & 0.025 & 0.172 & 0.172\\
			CJD17 & 6 & 6 & 6 & 6 & 0.038 & 0.055 & 0.072 & 0.072\\
			CJD18 & 6 & 6 & 6 & 6 & 0.037 & 0.047 & 0.028 & 0.047\\
			CJD19 & 6 & 6 & 6 & 6 & 0.033 & 0.033 & 0.047 & 0.047\\
			CJD20 & 6 & 6 & 6 & 6 & 0.037 & 0.022 & 0.021 & 0.037\\		
			CJD21 & 6 & 6 & 6 & 6 & 0.046 & 0.003 & 0.024 & 0.046\\
			CJD22 & 6 & 6 & 6 & 6 & 0.036 & 0.022 & 0.028 & 0.036\\
			CJD23 & 6 & 6 & 6 & 6 & 0.050 & 0.069 & 0.056 & 0.069\\
			CJD24 & 6 & 6 & 6 & 6 & 0.040 & 0.036 & 0.062 & 0.062\\
			CJD25 & 6 & 6 & 6 & 6 & 0.034 & 0.046 & 0.060 & 0.060\\	
			CJD26 & 6 & 6 & 6 & 6 & 0.035 & 0.041 & 0.033 & 0.041\\
			CJD27 & 6 & 6 & 6 & 6 & 0.057 & 0.034 & 0.034 & 0.057\\
			CJD28 & 6 & 6 & 6 & 6 & 0.048 & 0.037 & 0.035 & 0.048\\
			CJD29 & 6 & 6 & 6 & 6 & 0.043 & 0.032 & 0.050 & 0.050\\
			CJD30 & 6 & 6 & 6 & 6 & 0.039 & 49.857* & 0.063 & 0.063\\		
			CJD31 & 6 & 6 & 6 & 6 & 0.028 & 0.057 & 0.076 & 0.076\\
			\hline \Tstrut
			Control1 & 6 & 6 & 6 & 6 & 0.062 & 0.052 & 0.051 & 0.062\\
			Control2 & 3 & 2 & 3 & 1 & 0.000 & 0.024 & 0.000 & 0.024\\
			Control3 & 6 & 6 & 6 & 6 & 0.025 & 0.033 & 0.037 & 0.037\\
			Control4 & 6 & 6 & 6 & 6 & 0.019 & 0.033 & 0.043 & 0.043\\
			Control5 & 6 & 6 & 6 & 6 & 0.029 & 0.071 & 0.067 & 0.071\\
			Control6 & 6 & 6 & 6 & 6 & 0.026 & 0.043 & 0.034 & 0.043\\
			Control7 & 6 & 6 & 6 & 6 & 0.039 & 0.044 & 0.025 & 0.044\\
			Control8 & 6 & 6 & 6 & 6 & 0.009 & 0.014 & 0.029 & 0.029\\
			\hline \Tstrut
			All CJD & 179 & 179 & 179 & 179 & 0.073 & 0.078 & 0.172 & 0.172\\
			All controls & 45 & 44 & 45 & 43 & 0.062 & 0.071 & 0.067 & 0.071\\
			\bottomrule[2pt]
			\end{tabular}
		\end{adjustbox}
	\caption{\textbf{Overview of ddPCR single-nucleotide variant analysis.} Purified brain DNA from $31$ CJD and $8$ control individuals was analysed. Shown are the number of samples passing ddPCR quality control overall (\textit{analysed samples}) and per assay (\textit{analysed \dn}, \textit{analysed \ek}, \textit{analysed \pl}), together with the maximum estimated variant allele fraction (\textit{VAF}; \textit{max. VAF}) per assay and the maximum across assays (\textit{Max. VAF overall}). For the aggregate values overall \textit{Max. VAF overall} and \textit{All CJD}, the values for \ek were computed excluding CJD30, which was unexpectedly found to be germline heterozygous for \ek (marked with *). VAFs are reported as a fraction of total analysed DNA.}
	\label{tab:ddPCR-results}
\end{table}


\clearpage

% huge table with all values

\begin{table}[t]
	\captionsetup{font=small}
	\centering
	
	% Tighten horizontal & vertical spacing without breaking booktabs
	\setlength{\tabcolsep}{2.5pt}      % default ~6pt
	\renewcommand{\arraystretch}{0.80} % default 1.0
	\setlength{\aboverulesep}{0pt}
	\setlength{\belowrulesep}{0pt}
	
	% Fit within page text area (height guard keeps it on one page)
	\begin{adjustbox}{max totalsize={\textwidth}{0.9\textheight},center}
		\scriptsize % or \footnotesize; use \scriptsize for more rows on one page
		
		\begin{tabular}{@{}cllrrrrrr@{}}
			\toprule
		participant & histotype & mutation & basal ganglia & cerebellum & hippocampus & frontal cortex & substantia nigra & thalamus \\ 
\midrule\addlinespace[2.5pt]
CJD1 & MM1+2C & D178N & 0.015 (0-0.031) & 0.006 (0-0.015) & — & 0.015 (0-0.031) & 0.011 (0-0.024) & 0 (0-0) \\ 
CJD1 & MM1+2C & E200K & 0.032 (0.012-0.052) & 0.024 (0.007-0.041) & — & 0.012 (0-0.027) & 0.004 (0-0.013) & 0.046 (0-0.092) \\ 
CJD1 & MM1+2C & P102L & 0.021 (0.005-0.037) & 0.013 (0.002-0.025) & — & 0.032 (0.011-0.054) & 0.025 (0.008-0.042) & 0.023 (0.009-0.038) \\ 
CJD2 & MM1 & D178N & {\cellcolor[HTML]{E5E5E5}\textbf{0.067} (0.029–0.106)} & 0.022 (0-0.049) & 0 (0-0) & 0.025 (0.002-0.047) & 0.007 (0-0.015) & 0.003 (0-0.008) \\ 
CJD2 & MM1 & E200K & 0.026 (0-0.052) & 0.023 (0-0.051) & 0.019 (0-0.039) & 0.033 (0.008-0.059) & 0.023 (0.008-0.039) & {\cellcolor[HTML]{E5E5E5}{0.016 (0.008-0.025)}} \\ 
CJD2 & MM1 & P102L & 0.021 (0-0.043) & 0 (0-0) & 0.013 (0-0.032) & 0.044 (0.016-0.072) & 0.02 (0.002-0.037) & 0.001 (0.001-0.001) \\ 
CJD3 & atypical & D178N & 0.047 (0.023-0.07) & 0.028 (0.009-0.046) & 0.03 (0.015-0.044) & 0.009 (0-0.024) & {\cellcolor[HTML]{E5E5E5}{0.037 (0.021-0.053)}} & 0.043 (0.019-0.067) \\ 
CJD3 & atypical & E200K & {\cellcolor[HTML]{E5E5E5}{0.044 (0.023-0.065)}} & 0.04 (0.019-0.061) & 0.018 (0.008-0.028) & 0.043 (0.016-0.071) & {\cellcolor[HTML]{E5E5E5}{0.029 (0.017-0.041)}} & 0.045 (0.016-0.073) \\ 
CJD3 & atypical & P102L & 0.019 (0.006-0.033) & 0.021 (0.006-0.036) & 0.039 (0.022-0.057) & 0.053 (0.013-0.093) & {\cellcolor[HTML]{E5E5E5}{0.054 (0.036-0.073)}} & 0.013 (0-0.03) \\ 
CJD4 & VV2 & D178N & 0.027 (0.01-0.045) & 0.001 (0-0.001) & — & 0.021 (0-0.043) & — & {\cellcolor[HTML]{E5E5E5}{0.046 (0.028-0.065)}} \\ 
CJD4 & VV2 & E200K & 0.001 (0.001-0.001) & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} & — & 0 (0-0.001) & — & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.002)}} \\ 
CJD4 & VV2 & P102L & 0.049 (0.026-0.072) & 0.036 (0.018-0.054) & — & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} & — & 0.04 (0.023-0.058) \\ 
CJD5 & MV2K+C & D178N & 0.003 (0-0.007) & {\cellcolor[HTML]{E5E5E5}{0.02 (0.008-0.032)}} & {\cellcolor[HTML]{E5E5E5}{0.037 (0.021-0.052)}} & 0.003 (0-0.008) & 0.001 (0-0.004) & {\cellcolor[HTML]{E5E5E5}{0.042 (0.027-0.056)}} \\ 
CJD5 & MV2K+C & E200K & 0.006 (0.001-0.012) & 0.012 (0.004-0.021) & 0.019 (0.007-0.031) & 0.014 (0.005-0.023) & 0.003 (0-0.006) & {\cellcolor[HTML]{E5E5E5}{0.022 (0.012-0.032)}} \\ 
CJD5 & MV2K+C & P102L & 0.008 (0.001-0.016) & {\cellcolor[HTML]{E5E5E5}{0.042 (0.026-0.059)}} & {\cellcolor[HTML]{E5E5E5}{0.043 (0.025-0.062)}} & 0.001 (0-0.004) & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.002)}} & 0.009 (0.002-0.016) \\ 
CJD6 & MV2C & D178N & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.002)}} & 0.005 (0.001-0.009) & 0.009 (0.002-0.015) & {\cellcolor[HTML]{E5E5E5}{0.023 (0.01-0.036)}} & 0.001 (0-0.004) & 0.001 (0-0.001) \\ 
CJD6 & MV2C & E200K & {\cellcolor[HTML]{E5E5E5}{0.047 (0.029-0.065)}} & {\cellcolor[HTML]{E5E5E5}{0.012 (0.006-0.018)}} & 0.006 (0.001-0.011) & {\cellcolor[HTML]{E5E5E5}{0.065 (0.045-0.085)}} & 0.002 (0-0.005) & {\cellcolor[HTML]{E5E5E5}{0.001 (0-0.001)}} \\ 
CJD6 & MV2C & P102L & 0.006 (0-0.012) & {\cellcolor[HTML]{E5E5E5}{0.009 (0.003-0.014)}} & 0.001 (0-0.005) & {\cellcolor[HTML]{E5E5E5}{0.056 (0.035-0.076)}} & 0.002 (0-0.008) & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} \\ 
CJD7 & atypical & D178N & 0 (0-0) & 0.02 (0-0.051) & 0 (0-0) & 0.01 (0-0.034) & 0.008 (0-0.02) & 0 (0-0.001) \\ 
CJD7 & atypical & E200K & 0.017 (0.003-0.031) & {\cellcolor[HTML]{E5E5E5}{0 (0-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.021 (0.006-0.036)}} & 0.009 (0-0.018) & 0.003 (0-0.008) & 0.016 (0.001-0.03) \\ 
CJD7 & atypical & P102L & {\cellcolor[HTML]{E5E5E5}{0.058 (0.031-0.084)}} & 0.028 (0.005-0.05) & {\cellcolor[HTML]{E5E5E5}{0.001 (0-0.001)}} & 0.033 (0.015-0.051) & 0 (0-0.001) & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} \\ 
CJD8 & MM1 & D178N & 0 (0-0.001) & {\cellcolor[HTML]{E5E5E5}{0.052 (0.035-0.069)}} & 0.017 (0.007-0.027) & 0 (0-0) & 0.014 (0.007-0.021) & 0.017 (0.008-0.025) \\ 
CJD8 & MM1 & E200K & 0 (0-0.001) & 0.028 (0-0.056) & {\cellcolor[HTML]{E5E5E5}{0.023 (0.012-0.035)}} & 0.001 (0-0.003) & 0.011 (0.004-0.017) & {\cellcolor[HTML]{E5E5E5}{0.044 (0.029-0.059)}} \\ 
CJD8 & MM1 & P102L & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.002)}} & 0.168 (0.085-0.252) & 0.052 (0.029-0.075) & 0.006 (0-0.013) & 0.001 (0-0.004) & {\cellcolor[HTML]{E5E5E5}{0.012 (0.004-0.021)}} \\ 
CJD9 & atypical & D178N & {\cellcolor[HTML]{E5E5E5}{0.033 (0.016-0.05)}} & 0.003 (0-0.007) & {\cellcolor[HTML]{E5E5E5}{0.022 (0.011-0.032)}} & 0.001 (0-0.002) & {\cellcolor[HTML]{E5E5E5}{0.033 (0.017-0.049)}} & 0.001 (0-0.003) \\ 
CJD9 & atypical & E200K & 0.019 (0.006-0.032) & 0.001 (0-0.004) & 0.02 (0.009-0.031) & 0.003 (0-0.005) & 0.03 (0.015-0.044) & 0.002 (0-0.005) \\ 
CJD9 & atypical & P102L & 0.027 (0.012-0.043) & 0.005 (0-0.011) & {\cellcolor[HTML]{E5E5E5}{0.038 (0.023-0.052)}} & {\cellcolor[HTML]{E5E5E5}{0.017 (0.006-0.028)}} & {\cellcolor[HTML]{E5E5E5}{0.047 (0.027-0.067)}} & {\cellcolor[HTML]{E5E5E5}{0.012 (0.003-0.022)}} \\ 
CJD10 & MM1+2C & D178N & 0.004 (0-0.009) & {\cellcolor[HTML]{E5E5E5}{0.023 (0.015-0.031)}} & {\cellcolor[HTML]{E5E5E5}{0.018 (0.009-0.027)}} & 0.031 (0.019-0.042) & {\cellcolor[HTML]{E5E5E5}{0.023 (0.01-0.035)}} & {\cellcolor[HTML]{E5E5E5}{0.043 (0.027-0.058)}} \\ 
CJD10 & MM1+2C & E200K & 0.001 (0-0.004) & 0.014 (0.007-0.021) & {\cellcolor[HTML]{E5E5E5}{0.026 (0.015-0.037)}} & {\cellcolor[HTML]{E5E5E5}{0.032 (0.019-0.044)}} & 0.012 (0.003-0.02) & 0.012 (0.004-0.02) \\ 
CJD10 & MM1+2C & P102L & 0 (0-0) & 0 (0-0) & {\cellcolor[HTML]{E5E5E5}{0.031 (0.018-0.044)}} & 0.005 (0-0.009) & {\cellcolor[HTML]{E5E5E5}{0.086 (0.06-0.112)}} & {\cellcolor[HTML]{E5E5E5}{0.069 (0.047-0.091)}} \\ 
CJD11 & MM1 & D178N & 0.017 (0-0.034) & 0.019 (0-0.043) & 0.025 (0.002-0.047) & 0.026 (0.004-0.047) & 0.019 (0-0.041) & 0.035 (0.008-0.061) \\ 
CJD11 & MM1 & E200K & 0.011 (0-0.027) & 0.019 (0-0.038) & 0.015 (0-0.033) & 0.011 (0-0.028) & 0.022 (0.004-0.039) & 0.013 (0-0.033) \\ 
CJD11 & MM1 & P102L & 0.043 (0.02-0.066) & 0.033 (0.013-0.053) & 0.02 (0.003-0.037) & 0.026 (0.01-0.042) & 0.033 (0.014-0.052) & {\cellcolor[HTML]{E5E5E5}{0.045 (0.024-0.066)}} \\ 
CJD12 & MM1 & D178N & 0.04 (0.007-0.073) & 0.017 (0-0.038) & 0.026 (0.005-0.048) & 0.022 (0.004-0.04) & 0.02 (0-0.045) & 0.004 (0-0.009) \\ 
CJD12 & MM1 & E200K & {\cellcolor[HTML]{E5E5E5}{0.049 (0.019-0.078)}} & 0.012 (0-0.026) & {\cellcolor[HTML]{E5E5E5}{0.044 (0.01-0.078)}} & {\cellcolor[HTML]{E5E5E5}{0.051 (0.019-0.083)}} & 0.008 (0-0.02) & 0.002 (0-0.007) \\ 
CJD12 & MM1 & P102L & {\cellcolor[HTML]{E5E5E5}{0.046 (0.026-0.067)}} & 0.035 (0.014-0.056) & 0.029 (0.012-0.045) & 0.028 (0.012-0.044) & 0.033 (0.014-0.051) & 0.021 (0.008-0.033) \\ 
CJD13 & MM1 & D178N & 0.016 (0-0.036) & 0.009 (0-0.023) & 0.01 (0-0.033) & 0.015 (0-0.037) & {\cellcolor[HTML]{E5E5E5}{\textbf{0.073} (0.031-0.115)}} & 0.004 (0-0.014) \\ 
CJD13 & MM1 & E200K & 0.031 (0.003-0.059) & {\cellcolor[HTML]{E5E5E5}{\textbf{0.078} (0.035-0.122)}} & 0.013 (0-0.028) & 0.011 (0-0.024) & 0.025 (0.002-0.049) & 0 (0-0) \\ 
CJD13 & MM1 & P102L & 0.022 (0.008-0.036) & 0 (0-0) & 0.034 (0.016-0.052) & 0.016 (0.001-0.03) & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0-0.001)}} \\ 
CJD14 & atypical & D178N & 0 (0-0) & 0 (0-0) & 0.015 (0-0.033) & 0.012 (0-0.027) & {\cellcolor[HTML]{E5E5E5}{0.033 (0.008-0.058)}} & 0.006 (0-0.015) \\ 
CJD14 & atypical & E200K & 0 (0-0.001) & 0.003 (0-0.01) & {\cellcolor[HTML]{E5E5E5}{0.035 (0.015-0.055)}} & 0.016 (0.003-0.029) & 0.003 (0-0.007) & 0.028 (0.011-0.045) \\ 
CJD14 & atypical & P102L & {\cellcolor[HTML]{E5E5E5}{0 (0-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0-0.001)}} & 0 (0-0) & 0 (0-0) & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} \\ 
CJD15 & MM1+2C & D178N & 0.026 (0.005-0.048) & 0.024 (0.002-0.045) & 0.018 (0-0.036) & 0.021 (0-0.043) & 0.018 (0-0.037) & 0.035 (0.008-0.062) \\ 
CJD15 & MM1+2C & E200K & 0.008 (0-0.02) & 0.009 (0-0.022) & 0.005 (0-0.017) & {\cellcolor[HTML]{E5E5E5}{0.042 (0.015-0.068)}} & 0.008 (0-0.021) & 0.015 (0-0.034) \\ 
CJD15 & MM1+2C & P102L & 0.032 (0.015-0.049) & 0.008 (0-0.017) & 0.019 (0.007-0.03) & 0.038 (0.016-0.059) & 0.131 (0.052-0.21) & {\cellcolor[HTML]{E5E5E5}{0.076 (0.039-0.113)}} \\ 
CJD16 & MM1 & D178N & — & — & — & 0.054 (0.013-0.094) & — & — \\ 
CJD16 & MM1 & E200K & — & — & — & 0.025 (0-0.052) & — & — \\ 
CJD16 & MM1 & P102L & — & — & — & 0.172 (0.094-0.251) & — & — \\ 
CJD17 & MM1+2C & D178N & 0.011 (0-0.027) & {\cellcolor[HTML]{E5E5E5}{0.038 (0.011-0.065)}} & {\cellcolor[HTML]{E5E5E5}{0.036 (0.022-0.049)}} & 0.019 (0-0.038) & 0.02 (0-0.04) & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} \\ 
CJD17 & MM1+2C & E200K & 0.016 (0-0.036) & 0.036 (0.009-0.063) & {\cellcolor[HTML]{E5E5E5}{0.021 (0.011-0.031)}} & 0.018 (0-0.037) & 0.055 (0.018-0.092) & 0.001 (0-0.001) \\ 
CJD17 & MM1+2C & P102L & 0.018 (0.005-0.031) & 0.03 (0.009-0.051) & {\cellcolor[HTML]{E5E5E5}{0.046 (0.028-0.063)}} & 0.039 (0.016-0.061) & 0.042 (0.01-0.074) & {\cellcolor[HTML]{E5E5E5}{0.072 (0.048-0.097)}} \\ 
CJD18 & MV2K+C & D178N & 0.011 (0.004-0.018) & 0.037 (0.009-0.066) & 0.024 (0.009-0.04) & 0.023 (0.01-0.036) & 0.026 (0.011-0.041) & 0.015 (0.003-0.027) \\ 
CJD18 & MV2K+C & E200K & {\cellcolor[HTML]{E5E5E5}{0.025 (0.013-0.037)}} & 0.032 (0.006-0.058) & 0.018 (0.004-0.032) & 0.019 (0.006-0.031) & 0.047 (0.004-0.09) & 0.022 (0.008-0.036) \\ 
CJD18 & MV2K+C & P102L & 0.001 (0-0.004) & 0.028 (0.002-0.053) & 0.012 (0.001-0.024) & 0.002 (0-0.007) & {\cellcolor[HTML]{E5E5E5}{0.028 (0.014-0.042)}} & 0.01 (0-0.02) \\ 
CJD19 & atypical & D178N & 0.014 (0.005-0.022) & 0.006 (0-0.012) & 0.027 (0.01-0.045) & 0.032 (0.012-0.052) & 0.033 (0.019-0.048) & {\cellcolor[HTML]{E5E5E5}{0.02 (0.01-0.031)}} \\ 
CJD19 & atypical & E200K & 0.017 (0.006-0.027) & 0.003 (0-0.008) & 0.024 (0.009-0.038) & 0.017 (0.003-0.031) & 0.033 (0.017-0.048) & {\cellcolor[HTML]{E5E5E5}{0.033 (0.02-0.046)}} \\ 
CJD19 & atypical & P102L & {\cellcolor[HTML]{E5E5E5}{0.028 (0.014-0.042)}} & 0.014 (0.005-0.023) & 0.013 (0.002-0.025) & 0.019 (0.006-0.033) & 0.029 (0.014-0.045) & 0.047 (0.029-0.065) \\ 
CJD20 & MM1 & D178N & {\cellcolor[HTML]{E5E5E5}{0.037 (0.023-0.051)}} & 0.006 (0-0.016) & 0.006 (0-0.013) & 0.007 (0-0.016) & 0.007 (0-0.017) & 0.011 (0.001-0.021) \\ 
CJD20 & MM1 & E200K & 0.022 (0.008-0.036) & 0 (0-0) & 0 (0-0) & 0.002 (0-0.006) & 0.007 (0-0.019) & 0 (0-0) \\ 
CJD20 & MM1 & P102L & 0.021 (0.005-0.038) & 0.006 (0-0.013) & 0.01 (0-0.019) & 0.002 (0-0.005) & 0.013 (0-0.028) & 0.011 (0.002-0.02) \\ 
CJD21 & MM1 & D178N & 0.046 (0.022-0.071) & 0.017 (0.004-0.03) & 0.007 (0-0.015) & 0.014 (0.001-0.026) & 0.013 (0.003-0.023) & 0.004 (0-0.014) \\ 
CJD21 & MM1 & E200K & 0.001 (0-0.001) & 0.001 (0-0.001) & {\cellcolor[HTML]{E5E5E5}{0 (0-0.001)}} & 0.001 (0-0.001) & {\cellcolor[HTML]{E5E5E5}{0 (0-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.003 (0.003-0.004)}} \\ 
CJD21 & MM1 & P102L & 0.024 (0.006-0.043) & 0.004 (0-0.011) & 0.01 (0-0.02) & 0.01 (0-0.023) & 0.007 (0.001-0.013) & 0.016 (0.001-0.03) \\ 
CJD22 & MM1 & D178N & 0.007 (0-0.016) & 0.005 (0-0.012) & 0.036 (0.01-0.062) & 0.002 (0-0.007) & 0.004 (0-0.011) & 0.002 (0-0.008) \\ 
CJD22 & MM1 & E200K & 0.012 (0.001-0.022) & 0.005 (0-0.011) & 0.022 (0.002-0.043) & 0.016 (0.005-0.027) & 0.012 (0.002-0.021) & 0.016 (0.005-0.028) \\ 
CJD22 & MM1 & P102L & 0.014 (0.003-0.025) & 0.006 (0-0.014) & 0.028 (0.007-0.049) & 0.011 (0.001-0.021) & 0.012 (0.002-0.022) & 0.022 (0.006-0.038) \\ 
CJD23 & VV1 & D178N & {\cellcolor[HTML]{E5E5E5}{0.039 (0.021-0.057)}} & 0.03 (0.011-0.049) & {\cellcolor[HTML]{E5E5E5}{0.044 (0.026-0.061)}} & {\cellcolor[HTML]{E5E5E5}{0.029 (0.016-0.043)}} & {\cellcolor[HTML]{E5E5E5}{0.05 (0.031-0.069)}} & {\cellcolor[HTML]{E5E5E5}{0.039 (0.023-0.055)}} \\ 
CJD23 & VV1 & E200K & 0.025 (0.012-0.038) & {\cellcolor[HTML]{E5E5E5}{\textbf{0.069} (0.041-0.098)}} & {\cellcolor[HTML]{E5E5E5}{0.027 (0.015-0.039)}} & {\cellcolor[HTML]{E5E5E5}{0.03 (0.016-0.043)}} & 0.023 (0.011-0.035) & 0.021 (0.01-0.033) \\ 
CJD23 & VV1 & P102L & {\cellcolor[HTML]{E5E5E5}{0.056 (0.034-0.078)}} & 0.032 (0.01-0.053) & {\cellcolor[HTML]{E5E5E5}{0.043 (0.027-0.06)}} & 0.022 (0.011-0.034) & {\cellcolor[HTML]{E5E5E5}{0.034 (0.02-0.048)}} & {\cellcolor[HTML]{E5E5E5}{0.047 (0.022-0.072)}} \\ 
CJD24 & MM2T & D178N & {\cellcolor[HTML]{E5E5E5}{0.04 (0.023-0.056)}} & 0.013 (0.004-0.022) & {\cellcolor[HTML]{E5E5E5}{0.026 (0.014-0.039)}} & {\cellcolor[HTML]{E5E5E5}{0.027 (0.015-0.038)}} & {\cellcolor[HTML]{E5E5E5}{0.024 (0.015-0.034)}} & {\cellcolor[HTML]{E5E5E5}{0.028 (0.016-0.04)}} \\ 
CJD24 & MM2T & E200K & 0.023 (0.011-0.036) & 0.016 (0.008-0.024) & {\cellcolor[HTML]{E5E5E5}{0.028 (0.014-0.041)}} & {\cellcolor[HTML]{E5E5E5}{0.021 (0.012-0.031)}} & {\cellcolor[HTML]{E5E5E5}{0.019 (0.01-0.027)}} & {\cellcolor[HTML]{E5E5E5}{0.036 (0.023-0.049)}} \\ 
CJD24 & MM2T & P102L & {\cellcolor[HTML]{E5E5E5}{0.062 (0.035-0.089)}} & 0.022 (0.011-0.034) & {\cellcolor[HTML]{E5E5E5}{0.044 (0.024-0.063)}} & {\cellcolor[HTML]{E5E5E5}{0.04 (0.024-0.057)}} & {\cellcolor[HTML]{E5E5E5}{0.041 (0.026-0.055)}} & {\cellcolor[HTML]{E5E5E5}{0.031 (0.018-0.045)}} \\ 
CJD25 & MM2T & D178N & 0.016 (0.005-0.028) & {\cellcolor[HTML]{E5E5E5}{0.032 (0.016-0.047)}} & {\cellcolor[HTML]{E5E5E5}{0.034 (0.021-0.047)}} & {\cellcolor[HTML]{E5E5E5}{0.021 (0.011-0.031)}} & {\cellcolor[HTML]{E5E5E5}{0.019 (0.01-0.028)}} & {\cellcolor[HTML]{E5E5E5}{0.025 (0.014-0.035)}} \\ 
CJD25 & MM2T & E200K & {\cellcolor[HTML]{E5E5E5}{0.046 (0.028-0.064)}} & 0.017 (0.006-0.028) & 0.018 (0.008-0.028) & {\cellcolor[HTML]{E5E5E5}{0.022 (0.013-0.032)}} & 0.022 (0.01-0.033) & {\cellcolor[HTML]{E5E5E5}{0.024 (0.013-0.036)}} \\ 
CJD25 & MM2T & P102L & 0.06 (0.029-0.09) & 0.036 (0.019-0.053) & {\cellcolor[HTML]{E5E5E5}{0.036 (0.021-0.052)}} & {\cellcolor[HTML]{E5E5E5}{0.031 (0.019-0.043)}} & {\cellcolor[HTML]{E5E5E5}{0.031 (0.017-0.045)}} & {\cellcolor[HTML]{E5E5E5}{0.034 (0.02-0.047)}} \\ 
CJD26 & MM2T & D178N & {\cellcolor[HTML]{E5E5E5}{0.025 (0.015-0.035)}} & {\cellcolor[HTML]{E5E5E5}{0.028 (0.016-0.04)}} & {\cellcolor[HTML]{E5E5E5}{0.027 (0.018-0.037)}} & {\cellcolor[HTML]{E5E5E5}{0.026 (0.016-0.036)}} & {\cellcolor[HTML]{E5E5E5}{0.03 (0.02-0.04)}} & {\cellcolor[HTML]{E5E5E5}{0.035 (0.024-0.047)}} \\ 
CJD26 & MM2T & E200K & {\cellcolor[HTML]{E5E5E5}{0.027 (0.016-0.038)}} & {\cellcolor[HTML]{E5E5E5}{0.041 (0.027-0.056)}} & {\cellcolor[HTML]{E5E5E5}{0.02 (0.012-0.028)}} & {\cellcolor[HTML]{E5E5E5}{0.037 (0.024-0.049)}} & {\cellcolor[HTML]{E5E5E5}{0.017 (0.01-0.024)}} & 0.02 (0.011-0.029) \\ 
CJD26 & MM2T & P102L & {\cellcolor[HTML]{E5E5E5}{0.033 (0.021-0.044)}} & 0.014 (0.006-0.022) & {\cellcolor[HTML]{E5E5E5}{0.02 (0.011-0.029)}} & {\cellcolor[HTML]{E5E5E5}{0.029 (0.018-0.04)}} & {\cellcolor[HTML]{E5E5E5}{0.02 (0.011-0.029)}} & 0.015 (0.007-0.023) \\ 
CJD27 & MV2K & D178N & 0.057 (0.028-0.086) & 0.051 (0.031-0.07) & 0.029 (0.016-0.042) & 0.029 (0.016-0.041) & 0.022 (0.014-0.031) & 0.032 (0.017-0.046) \\ 
CJD27 & MV2K & E200K & 0.034 (0.012-0.055) & 0.028 (0.013-0.043) & 0.02 (0.009-0.031) & 0.021 (0.01-0.031) & {\cellcolor[HTML]{E5E5E5}{0.023 (0.015-0.032)}} & {\cellcolor[HTML]{E5E5E5}{0.028 (0.016-0.04)}} \\ 
CJD27 & MV2K & P102L & 0.034 (0.012-0.055) & 0.022 (0.009-0.035) & {\cellcolor[HTML]{E5E5E5}{0.023 (0.011-0.035)}} & 0.017 (0.008-0.025) & {\cellcolor[HTML]{E5E5E5}{0.02 (0.011-0.028)}} & 0.014 (0.005-0.024) \\ 
CJD28 & MV2K & D178N & 0.029 (0.015-0.042) & 0.027 (0.016-0.038) & 0.018 (0.008-0.029) & 0.023 (0.011-0.035) & 0.032 (0.018-0.046) & 0.048 (0.026-0.071) \\ 
CJD28 & MV2K & E200K & {\cellcolor[HTML]{E5E5E5}{0.03 (0.016-0.044)}} & 0.02 (0.01-0.03) & {\cellcolor[HTML]{E5E5E5}{0.027 (0.015-0.04)}} & 0.032 (0.016-0.047) & {\cellcolor[HTML]{E5E5E5}{0.037 (0.019-0.054)}} & 0.034 (0.013-0.054) \\ 
CJD28 & MV2K & P102L & {\cellcolor[HTML]{E5E5E5}{0.034 (0.019-0.05)}} & {\cellcolor[HTML]{E5E5E5}{0.024 (0.012-0.035)}} & {\cellcolor[HTML]{E5E5E5}{0.025 (0.013-0.038)}} & {\cellcolor[HTML]{E5E5E5}{0.029 (0.015-0.042)}} & {\cellcolor[HTML]{E5E5E5}{0.035 (0.02-0.05)}} & 0.031 (0.012-0.049) \\ 
CJD29 & MV2K & D178N & {\cellcolor[HTML]{E5E5E5}{0.043 (0.027-0.059)}} & {\cellcolor[HTML]{E5E5E5}{0.031 (0.02-0.042)}} & 0.042 (0.021-0.063) & 0.039 (0.024-0.053) & 0.031 (0.019-0.043) & 0.022 (0.012-0.031) \\ 
CJD29 & MV2K & E200K & {\cellcolor[HTML]{E5E5E5}{0.032 (0.018-0.046)}} & 0.015 (0.008-0.022) & 0.011 (0-0.022) & 0.017 (0.007-0.026) & 0.02 (0.011-0.03) & {\cellcolor[HTML]{E5E5E5}{0.028 (0.017-0.038)}} \\ 
CJD29 & MV2K & P102L & {\cellcolor[HTML]{E5E5E5}{0.028 (0.015-0.041)}} & 0.011 (0.004-0.018) & {\cellcolor[HTML]{E5E5E5}{0.05 (0.026-0.074)}} & {\cellcolor[HTML]{E5E5E5}{0.025 (0.013-0.038)}} & {\cellcolor[HTML]{E5E5E5}{0.023 (0.012-0.033)}} & {\cellcolor[HTML]{E5E5E5}{0.022 (0.012-0.033)}} \\ 
CJD30 & VV2 & D178N & {\cellcolor[HTML]{E5E5E5}{0.031 (0.017-0.045)}} & 0.027 (0.011-0.044) & 0.024 (0.012-0.036) & {\cellcolor[HTML]{E5E5E5}{0.029 (0.015-0.042)}} & 0.021 (0.008-0.035) & 0.039 (0.024-0.055) \\ 
CJD30 & VV2 & E200K & {\cellcolor[HTML]{E5E5E5}{49.522 (48.811-50.234)}} & {\cellcolor[HTML]{E5E5E5}{49.471 (48.755-50.187)}} & {\cellcolor[HTML]{E5E5E5}{48.939 (48.194-49.684)}} & {\cellcolor[HTML]{E5E5E5}{49.857 (49.148-50.566)}} & {\cellcolor[HTML]{E5E5E5}{49.48 (48.792-50.167)}} & {\cellcolor[HTML]{E5E5E5}{49.444 (48.693-50.195)}} \\ 
CJD30 & VV2 & P102L & 0.061 (0.04-0.082) & 0.063 (0.037-0.089) & {\cellcolor[HTML]{E5E5E5}{0.042 (0.027-0.057)}} & 0.045 (0.026-0.064) & {\cellcolor[HTML]{E5E5E5}{0.044 (0.028-0.059)}} & {\cellcolor[HTML]{E5E5E5}{0.039 (0.026-0.052)}} \\ 
CJD31 & VV2 & D178N & 0.019 (0.008-0.029) & 0.016 (0.006-0.025) & 0.024 (0.013-0.036) & 0.024 (0.01-0.039) & {\cellcolor[HTML]{E5E5E5}{0.028 (0.017-0.039)}} & 0.026 (0.014-0.037) \\ 
CJD31 & VV2 & E200K & {\cellcolor[HTML]{E5E5E5}{0.032 (0.02-0.045)}} & {\cellcolor[HTML]{E5E5E5}{0.024 (0.013-0.035)}} & {\cellcolor[HTML]{E5E5E5}{0.025 (0.016-0.034)}} & {\cellcolor[HTML]{E5E5E5}{0.057 (0.035-0.079)}} & {\cellcolor[HTML]{E5E5E5}{0.021 (0.013-0.029)}} & {\cellcolor[HTML]{E5E5E5}{0.053 (0.036-0.069)}} \\ 
CJD31 & VV2 & P102L & {\cellcolor[HTML]{E5E5E5}{0.042 (0.027-0.057)}} & 0.033 (0.018-0.048) & {\cellcolor[HTML]{E5E5E5}{0.028 (0.017-0.039)}} & {\cellcolor[HTML]{E5E5E5}{0.053 (0.032-0.073)}} & {\cellcolor[HTML]{E5E5E5}{0.039 (0.027-0.051)}} & {\cellcolor[HTML]{E5E5E5}{0.076 (0.054-0.099)}} \\ 
Control1 & control & D178N & 0.011 (0.002-0.021) & 0.062 (0.015-0.11) & 0.005 (0-0.011) & 0.003 (0-0.006) & {\cellcolor[HTML]{E5E5E5}{0.02 (0.011-0.029)}} & {\cellcolor[HTML]{E5E5E5}{0.031 (0.017-0.046)}} \\ 
Control1 & control & E200K & 0.018 (0.006-0.03) & 0.052 (0.022-0.082) & 0.011 (0.003-0.019) & 0.004 (0-0.007) & {\cellcolor[HTML]{E5E5E5}{0.021 (0.01-0.031)}} & 0.019 (0.007-0.03) \\ 
Control1 & control & P102L & 0.012 (0.001-0.022) & 0.044 (0.004-0.084) & 0.003 (0-0.008) & 0.005 (0-0.011) & {\cellcolor[HTML]{E5E5E5}{0.03 (0.018-0.041)}} & {\cellcolor[HTML]{E5E5E5}{0.051 (0.032-0.07)}} \\ 
Control2 & control & D178N & — & 0 (0-0) & — & — & — & 0 (0-0) \\ 
Control2 & control & E200K & — & 0.003 (0-0.006) & — & {\cellcolor[HTML]{E5E5E5}{0.024 (0.01-0.039)}} & — & 0.006 (0-0.012) \\ 
Control2 & control & P102L & — & 0 (0-0) & — & — & — & — \\ 
Control3 & control & D178N & 0.004 (0-0.012) & 0.003 (0-0.01) & {\cellcolor[HTML]{E5E5E5}{0.025 (0.015-0.035)}} & 0.005 (0-0.012) & 0.008 (0-0.019) & 0.02 (0-0.068) \\ 
Control3 & control & E200K & 0.01 (0-0.022) & 0.005 (0-0.013) & {\cellcolor[HTML]{E5E5E5}{0.022 (0.012-0.033)}} & 0.003 (0-0.009) & {\cellcolor[HTML]{E5E5E5}{0.033 (0.013-0.052)}} & 0.005 (0-0.016) \\ 
Control3 & control & P102L & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.037 (0.023-0.051)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.002)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} \\ 
Control4 & control & D178N & 0.005 (0-0.011) & 0.014 (0-0.028) & 0.019 (0-0.038) & 0.01 (0.002-0.018) & 0.01 (0-0.02) & 0.008 (0-0.017) \\ 
Control4 & control & E200K & 0.015 (0.006-0.023) & 0 (0-0) & 0.033 (0.01-0.057) & 0.01 (0.002-0.018) & 0.007 (0-0.014) & {\cellcolor[HTML]{E5E5E5}{0.02 (0.006-0.034)}} \\ 
Control4 & control & P102L & {\cellcolor[HTML]{E5E5E5}{0.007 (0.001-0.013)}} & 0.018 (0-0.036) & {\cellcolor[HTML]{E5E5E5}{0.043 (0.019-0.067)}} & 0 (0-0) & {\cellcolor[HTML]{E5E5E5}{0.028 (0.016-0.041)}} & {\cellcolor[HTML]{E5E5E5}{0.021 (0.007-0.035)}} \\ 
Control5 & control & D178N & 0.003 (0-0.011) & 0.01 (0-0.022) & 0.012 (0-0.025) & 0.029 (0.011-0.046) & 0.008 (0-0.027) & 0.01 (0-0.025) \\ 
Control5 & control & E200K & 0.005 (0-0.012) & 0.007 (0-0.019) & 0.061 (0.032-0.091) & {\cellcolor[HTML]{E5E5E5}{\textbf{0.071} (0.045-0.096)}} & 0.012 (0.002-0.021) & 0.016 (0-0.033) \\ 
Control5 & control & P102L & 0.009 (0-0.019) & 0.026 (0.006-0.047) & {\cellcolor[HTML]{E5E5E5}{0.019 (0.009-0.03)}} & {\cellcolor[HTML]{E5E5E5}{0.067 (0.042-0.092)}} & 0 (0-0) & {\cellcolor[HTML]{E5E5E5}{0.013 (0.003-0.023)}} \\ 
Control6 & control & D178N & 0.026 (0-0.065) & 0.018 (0-0.044) & 0.008 (0-0.027) & 0.025 (0.004-0.045) & {\cellcolor[HTML]{E5E5E5}{0.019 (0.01-0.029)}} & {\cellcolor[HTML]{E5E5E5}{0.019 (0.01-0.027)}} \\ 
Control6 & control & E200K & {\cellcolor[HTML]{E5E5E5}{0.024 (0.009-0.04)}} & {\cellcolor[HTML]{E5E5E5}{0.043 (0.014-0.072)}} & 0.005 (0-0.011) & 0.007 (0-0.015) & {\cellcolor[HTML]{E5E5E5}{0.034 (0.021-0.048)}} & {\cellcolor[HTML]{E5E5E5}{0.04 (0.027-0.052)}} \\ 
Control6 & control & P102L & {\cellcolor[HTML]{E5E5E5}{0.001 (0-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0.001-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.001 (0-0.001)}} & {\cellcolor[HTML]{E5E5E5}{0.034 (0.02-0.048)}} & {\cellcolor[HTML]{E5E5E5}{0.032 (0.021-0.043)}} \\ 
Control7 & control & D178N & 0.015 (0.001-0.029) & 0.007 (0-0.017) & {\cellcolor[HTML]{E5E5E5}{0.035 (0.013-0.057)}} & 0.003 (0-0.009) & {\cellcolor[HTML]{E5E5E5}{0.039 (0.018-0.06)}} & 0.015 (0-0.031) \\ 
Control7 & control & E200K & 0.013 (0.001-0.026) & 0.014 (0-0.028) & 0.01 (0-0.021) & 0.005 (0-0.014) & 0.044 (0.022-0.065) & 0.031 (0.011-0.05) \\ 
Control7 & control & P102L & 0.011 (0.001-0.02) & 0.013 (0-0.027) & 0.025 (0.007-0.043) & 0.01 (0-0.021) & 0.022 (0.007-0.037) & 0.023 (0.005-0.04) \\ 
Control8 & control & D178N & 0.007 (0-0.018) & 0.003 (0-0.008) & 0.001 (0-0.004) & 0.007 (0-0.015) & 0.009 (0.003-0.016) & 0.006 (0-0.012) \\ 
Control8 & control & E200K & 0.004 (0-0.014) & 0.014 (0.004-0.025) & 0.005 (0-0.01) & 0.005 (0-0.012) & 0.007 (0.001-0.014) & 0.008 (0.001-0.015) \\ 
Control8 & control & P102L & 0.023 (0.008-0.037) & 0.004 (0-0.011) & 0.011 (0.003-0.019) & 0.018 (0.005-0.031) & {\cellcolor[HTML]{E5E5E5}{0.029 (0.016-0.043)}} & 0.014 (0.004-0.024) \\ 
			\bottomrule
		\end{tabular}
		
	\end{adjustbox}
	
	\caption{\textbf{Fractional abundance (FA) of \textit{PRNP} point mutation signal across brain regions}, quantified by ddPCR and reported as percentages. Bold type indicates samples in which the mean FA, but not the lower bound of the 95\% confidence interval, surpassed the limit of detection (LoD). Grey shading denotes values exceeding the respective limit of blank (LoB).}

	\label{tab:ddpcr_results}
\end{table}

\clearpage

% Sequencing metrics table

\begin{table}[t]
	\centering
	\resizebox{\textwidth}{!}{
\begin{tabular}{llrrrrrrr}
	\toprule
	\multirow{2}{*}{Sample} & \multirow{2}{*}{Group}
	& \multicolumn{2}{c}{Reads}
	& \multicolumn{5}{c}{\textit{PRNP} coding region coverage} \\
	\cmidrule(lr){3-4}\cmidrule(lr){5-9}
	& & Primary mapped (n) & On-target (\%)
	& Mean depth (\xfold) & P20 depth (\xfold) & Fold-80
	& Bases $\geq$100\xfold (\%) & Bases $\geq$500\xfold (\%) \\
	\midrule
	CJD1 & CJD & 2971083 & 82.3 & 4874 & 4492 & 1.08 & 100.0 & 100.0\\
	CJD2 & CJD & 2707989 & 82.2 & 4749 & 4500 & 1.05 & 100.0 & 100.0\\
	CJD3 & CJD & 2418307 & 70.3 & 6854 & 5627 & 1.22 & 100.0 & 100.0\\
	CJD4 & CJD & 2367843 & 64.1 & 4132 & 3412 & 1.21 & 100.0 & 100.0\\
	CJD5 & CJD & 2418541 & 0.3 & 264 & 229 & 1.15 & 100.0 & 0.0\\
	CJD6 & CJD & 2879603 & 69.0 & 13997 & 13340 & 1.05 & 100.0 & 100.0\\
	CJD7 & CJD & 1201860 & 65.8 & 4418 & 3853 & 1.15 & 100.0 & 100.0\\
	CJD8 & CJD & 479340 & 3.4 & 533 & 461 & 1.16 & 100.0 & 58.7\\
	CJD11 & CJD & 1799120 & 67.3 & 5119 & 4498 & 1.14 & 100.0 & 100.0\\
	CJD12 & CJD & 764317 & 21.7 & 1204 & 1036 & 1.16 & 100.0 & 100.0\\
	CJD13 & CJD & 2332310 & 84.4 & 5267 & 4876 & 1.08 & 100.0 & 100.0\\
	CJD14 & CJD & 2028669 & 0.4 & 100 & 75 & 1.34 & 48.7 & 0.0\\
	CJD15 & CJD & 1757196 & 0.7 & 157 & 133 & 1.18 & 100.0 & 0.0\\
	CJD18 & CJD & 1943421 & 0.6 & 175 & 113 & 1.55 & 96.5 & 0.0\\
	CJD19 & CJD & 1030223 & 5.8 & 1114 & 909 & 1.23 & 100.0 & 100.0\\
	CJD21 & CJD & 847271 & 80.1 & 3905 & 3467 & 1.13 & 100.0 & 100.0\\
	CJD22 & CJD & 2656868 & 73.7 & 6299 & 5990 & 1.05 & 100.0 & 100.0\\
	CJD23 & CJD & 2256474 & 73.7 & 9503 & 8676 & 1.09 & 100.0 & 100.0\\
	CJD24 & CJD & 1822237 & 0.5 & 434 & 331 & 1.31 & 100.0 & 33.7\\
	CJD25 & CJD & 2513891 & 61.0 & 12492 & 9277 & 1.35 & 100.0 & 100.0\\
	CJD26 & CJD & 649606 & 11.9 & 699 & 560 & 1.25 & 100.0 & 94.2\\
	CJD27 & CJD & 5251449 & 77.0 & 8707 & 8454 & 1.03 & 100.0 & 100.0\\
	CJD28 & CJD & 816675 & 4.1 & 676 & 526 & 1.28 & 100.0 & 85.6\\
	CJD29 & CJD & 795483 & 6.4 & 1468 & 1374 & 1.07 & 100.0 & 100.0\\
	CJD30 & CJD & 1214681 & 3.3 & 840 & 759 & 1.11 & 100.0 & 100.0\\
	CJD31 & CJD & 550327 & 1.6 & 229 & 165 & 1.38 & 100.0 & 0.0\\
	\addlinespace
	Ctrl1 & control & 2570807 & 44.5 & 8908 & 8122 & 1.10 & 100.0 & 100.0\\
	Ctrl2 & control & 1295808 & 22.6 & 2656 & 2270 & 1.17 & 100.0 & 100.0\\
	Ctrl3 & control & 2701096 & 67.0 & 4380 & 4077 & 1.07 & 100.0 & 100.0\\
	Ctrl4 & control & 952052 & 2.8 & 666 & 551 & 1.21 & 100.0 & 90.2\\
	Ctrl5 & control & 1326561 & 62.7 & 5068 & 4772 & 1.06 & 100.0 & 100.0\\
	Ctrl7 & control & 882024 & 3.8 & 652 & 547 & 1.19 & 100.0 & 95.9\\
	\addlinespace
	Wildtype & spike-in & 2222988 & 77.8 & 5742 & 4985 & 1.15 & 100.0 & 100.0\\
	A117V 0.5\% spike-in & spike-in & 4751420 & 77.0 & 10050 & 7846 & 1.28 & 100.0 & 100.0\\
	A117V 1\% spike-in & spike-in & 3147044 & 81.6 & 4651 & 4182 & 1.11 & 100.0 & 100.0\\
	\bottomrule
	\end{tabular}
}
\vspace{0.5em}
\caption{\textbf{Sequencing quality metrics and \textit{PRNP} coding-region coverage}. For each sample, we report primary mapped reads, on-target read fraction, coverage statistics across the \textit{PRNP} coding region, including mean depth, P20 depth (20th percentile depth), Fold-80 (mean/P20; an indicator of coverage uniformity) and the percentage of coding bases covered at $\geq$100\xfold and $\geq$500\xfold.}
\end{table}


	
\end{document}
