
#this was generated by ChatGPT and only Steps 1-3 work properly


# Step 1: Extract reads overlapping the position of interest
samtools view -b -h "/add/2023-06-18_CJD_16_samples/312092_13-CJD30_G01_S16_R1_001.fastq.gz.picard.markedDup.recal.bam" "chr20:4699818-4699818" > "/add/2024-03-03_count_path_variants/CJD30_E200K_reads_of_interest.bam"

# Step 2: Filter reads supporting the alternate allele
bcftools mpileup -Ou -f /home/mcarta/databases/chr2_chr4_chr20.fasta "/add/2024-03-03_count_path_variants/CJD30_E200K_reads_of_interest.bam" | bcftools call -mv -Oz -o variants.vcf.gz
bcftools view -i 'ALT="A" && POS==4699818' variants.vcf.gz > alternate_allele.vcf

# Step 3: Count the number of samples in the VCF file
num_samples=$(bcftools query -l alternate_allele.vcf | wc -l)

# Step 4: Calculate allele frequency
total_reads=$(samtools view -c -F 4 "/add/2024-03-03_count_path_variants/CJD30_E200K_reads_of_interest.bam")
allele_reads=$(bcftools query -i 'ALT="A" && POS==4699818' -l alternate_allele.vcf | wc -l)

# the output of this is "1", both for REF="G" and ALT="A". It seems to have given the relative frequency, rather than the count, which is what I want.

allele_frequency=$(echo "scale=4; $allele_reads / $total_reads" | bc)

# Step 5: Create a file containing the computed variables
output_file="/add/2024-03-03_count_path_variants/output.txt"
echo "Total number of samples with the alternate allele: $num_samples" > "$output_file"
echo "Allele reads: $allele_reads" >> "$output_file"
echo "Allele frequency: $allele_frequency" >> "$output_file"

# Print the contents of the output file
cat "$output_file"
